cmd/algons/dnsaddrCmd.go

```diff
@@ -63,3 +78,94 @@ var dnsaddrTreeCmd = &cobra.Command{
 		}
 	},
 }
+var dnsaddrTreeDeleteCmd = &cobra.Command{
+	Use:   "delete",
+	Short: "Recursively resolves and deletes the dnsaddr entries of the given domain",
+	Long:  "Recursively resolves and deletes the dnsaddr entries of the given domain",
+	Run: func(cmd *cobra.Command, args []string) {
+		addr, err := multiaddr.NewMultiaddr(fmt.Sprintf("/dnsaddr/%s", dnsaddrDomain))
+		if err != nil {
+			fmt.Printf("unable to construct multiaddr for %s : %v\n", dnsaddrDomain, err)
+			return
+		}
+		controller := dnsaddr.NewMultiaddrDNSResolveController(secure, "")
+		cfZoneID, cfToken, err := getClouldflareCredentials()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "error getting DNS credentials: %v", err)
+			return
+		}
+		cloudflareDNS := cloudflare.NewDNS(cfZoneID, cfToken)
+		var recordsToDelete []cloudflare.DNSRecordResponseEntry
+		err = dnsaddr.Iterate(addr, controller, func(entryFrom multiaddr.Multiaddr, entries []multiaddr.Multiaddr) error {
+			domain, _ := entryFrom.ValueForProtocol(multiaddr.P_DNSADDR)
+			name := fmt.Sprintf("_dnsaddr.%s", domain)
+			fmt.Printf("listing records for %s\n", name)
+			records, err0 := cloudflareDNS.ListDNSRecord(context.Background(), "TXT", name, "", "", "", "")
+			if err0 != nil {
+				fmt.Printf("erroring listing dns records for %s %s\n", domain, err)
+				return err
+			}
+			for _, record := range records {
+				fmt.Printf("found record to delete %s:%s\n", record.Name, record.Content)
+				recordsToDelete = append(recordsToDelete, record)
+			}
+			return nil
+		})
+		if err != nil {
+			fmt.Printf("%s\n", err.Error())
+			return
+		}
+		err = checkedDelete(recordsToDelete, cloudflareDNS)
+		if err != nil {
+			fmt.Printf("error deleting records: %s\n", err)
+		}
+	},
+}
+
+var dnsaddrTreeCreateCmd = &cobra.Command{
+	Use:   "create",
+	Short: "Creates a tree of entries containing the multiaddrs at the provided root domain",
+	Long:  "Creates a tree of entries containing the multiaddrs at the provided root domain",
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(cmdMultiaddrs) == 0 {
+			fmt.Printf("must provide multiaddrs to put in the DNS records")
+			return
+		}
+		recordsPerEntry := 4
```

Comment:
Yeah, 4 is somewhat arbitrary I guess. I've been working with that as my initial record size purely based off of https://github.com/multiformats/multiaddr/blob/master/protocols/DNSADDR.md#caveats where they call out that they're using 4 entries per record.

Since this is pretty much only going to be used in ping-pong tests there shouldn't be any problem resolving larger sets of TXT entries. On the other hand it might be good to emulate whatever we plan on using for the production deployment--I'm not sure that we've decided how that will be setup yet.

I'm happy to increase to something like 50 as the default and move it to a CLI input so we can control it.